supysonic/db.py

```diff
@@ -572,6 +574,113 @@ def as_subsonic_station(self):
         return info
 
 
+@unique
+class PodcastStatus(Enum):
+    new = 1
+    downloading = 2
+    completed = 3
+    error = 4
+    deleted = 5
+    skipped = 6
+
+
+class PodcastChannel(db.Entity):
+    _table_ = "podcast_channel"
+
+    id = PrimaryKey(UUID, default=uuid4)
+    url = Required(str, unique=True)
+    title = Optional(str, nullable=True)
+    description = Optional(str, nullable=True)
+    cover_art = Optional(str, nullable=True)
+    original_image_url = Optional(str)
+    # Status params mirror PodcastStatus
+    status = Required(int, min=1, max=6, default=1)
+    error_message = Optional(str, nullable=True)
+    created = Required(datetime, precision=0, default=now)
+    last_fetched = Optional(datetime, precision=0, nullable=True)
+    episodes = Set(lambda: PodcastEpisode, lazy=True)
+
+    def soft_delete(self):
+        for ep in self.episodes:
+            ep.soft_delete()
+
+        self.status = PodcastStatus.deleted.value
+        self.error_message = None
+
+    def as_subsonic_channel(self, include_episodes=False):
+        info = dict(
+            id=self.id,
+            url=self.url,
+            status=PodcastStatus(self.status).name,
+        )
+
+        if self.title:
+            info["title"] = self.title
+        if self.description:
+            info["description"] = self.description
+        if self.error_message:
+            info["errorMessage"] = self.error_message
+
+        if include_episodes:
+            info["episode"] = [ep.as_subsonic_episode() for ep in self.episodes]
+        return info
+
+
+class PodcastEpisode(db.Entity):
+    _table_ = "podcast_episode"
+
+    id = PrimaryKey(UUID, default=uuid4)
+    channel = Required(PodcastChannel, column="channel_id")
+    # Location of the episode. Used to stream and download.
+    stream_url = Required(str)
+    # Path of file after it has been downloaded.
+    file_path = Optional(str, nullable=True)
+    title = Required(str)
+    description = Optional(str, nullable=True)
+    duration = Optional(str, nullable=True)
+    # Status params mirror PodcastStatus
+    status = Required(int, min=1, max=6, default=1)
+    publish_date = Optional(datetime, precision=0, default=now)
+    error_message = Optional(str, nullable=True)
+    created = Required(datetime, precision=0, default=now)
+    size = Optional(int, nullable=True)
+    suffix = Optional(str, nullable=True)
+    bitrate = Optional(str, nullable=True)
```

Comment:
I originally did this for flexibility since it's just used for display and not calculations. The RSS schema marks the data type as non-negative int, so I switched the field to int.